components/tag/src/index.tsx

```diff
@@ -16,34 +34,57 @@ import { typography } from '@govuk-react/lib';
  *
  */
 export const Tag = styled('strong')<TagProps>(
-  typography.font({ size: 16, weight: 'bold', lineHeight: 1.25 }),
+  typography.font({ size: 16, weight: 'bold', lineHeight: 1 }),
   {
     display: 'inline-block',
-    padding: '4px 8px',
-    paddingBottom: 1,
+    paddingTop: '5px',
+    paddingRight: '8px',
+    paddingBottom: '4px',
+    paddingLeft: '8px',
     outline: '2px solid transparent',
     outlineOffset: -2,
 
-    color: WHITE,
-    backgroundColor: BLUE,
     letterSpacing: 1,
 
     textDecoration: 'none',
     textTransform: 'uppercase',
   },
+  ({ tint = 'SOLID' }) => ({
+    backgroundColor: TAG_TINTS[tint].background,
+    color: TAG_TINTS[tint].text,
+  }),
   ({ inactive }) =>
     inactive
       ? {
-          backgroundColor: GREY_1,
+          backgroundColor: TAG_TINTS.GREY.background,
+          color: TAG_TINTS.GREY.text,
         }
-      : undefined
+      : undefined,
+  ({ backgroundColor }) => (backgroundColor ? { backgroundColor } : undefined),
+  ({ color }) => (color ? { color } : undefined)
 );
 export interface TagProps {
+  /**
+   * Deprecated: use grey tint instead
+   */
   inactive?: boolean;
+  /**
+   * Tag tint
+   */
+  tint?: keyof typeof TAG_TINTS;
+  /**
+   * Custom colour to override the tag background color
+   */
+  backgroundColor?: string;
+  /**
+   * Custom colour to override the tag text color
+   */
+  color?: string;
 }
 
 Tag.defaultProps = {
   inactive: false,
+  tint: 'SOLID',
```

Comment:
In usage, I'm not sure if we should be asking developers to make the tint prop all-upper or all-lower.

Either works, but I'd want to keep it consistent across govuk-react. Off the top of my head, I can't think of anywhere we're accepting enums at the moment, but we should check. If this is the first instance we're setting a precedence and I'd want to make sure we're happy.

@stevesims @Loque- @marksy any preference?

@domdomegg am I right in assuming your preference would be all-upper for the prop value?

Reply:
I'm not sure either.

I think my preference would actually be all-lower, but I don't really have a good reason for that (I think I've just used other libraries where it's been all lower and so it looks more 'normal to me').

Reason I went for all-upper here was copying the style of `Heading` which for its size prop states: "Visual size level, accepts: `XLARGE`, `LARGE`, `MEDIUM`, `SMALL`, `XL`, `L`, `M`, `S` or a numeric size that fits in the GDS font scale list".

But looking around other components, I think we also use all-lowercase e.g. for Grid Col's setWith: "Explicitly set column to width using value or descriptive string (`one-quarter`, `one-third`, `one-half`, `two-thirds`, `three-quarters`, `full`)"

Reply:
yep size prop on heading is all caps:

https://github.com/govuk-react/govuk-react/blob/01164b69da02cb78ce770d44a479f73e93d84c9b/packages/constants/src/typography.tsx#L152-L159

https://github.com/govuk-react/govuk-react/blob/01164b69da02cb78ce770d44a479f73e93d84c9b/components/heading/src/heading.tsx#L24

https://github.com/govuk-react/govuk-react/blob/01164b69da02cb78ce770d44a479f73e93d84c9b/components/heading/src/stories.tsx#L41

And setWidth prop on GridCol is all-lower:

https://github.com/govuk-react/govuk-react/blob/01164b69da02cb78ce770d44a479f73e93d84c9b/components/grid-col/src/index.tsx#L55

https://github.com/govuk-react/govuk-react/blob/01164b69da02cb78ce770d44a479f73e93d84c9b/packages/constants/src/spacing.tsx#L83-L88

https://github.com/govuk-react/govuk-react/blob/01164b69da02cb78ce770d44a479f73e93d84c9b/components/grid-col/src/stories.tsx#L27


Ok as we're already inconsistent I'm happy to keep this as-is as we can look in to consistent naming of enum props in future, I'll create a ticket.

Reply:
#1099